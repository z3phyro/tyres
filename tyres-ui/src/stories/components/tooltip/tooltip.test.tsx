import { expect, test } from "vitest";
import { render, screen } from "@solidjs/testing-library";
import Tooltip from "./tooltip";
import userEvent from "@testing-library/user-event";

test("Tooltip renders", () => {
  const { getByRole } = render(() => <Tooltip content="Test content">
    <button>Hover me</button>
  </Tooltip>);

  const tooltipElement = getByRole("tooltip");
  expect(tooltipElement).toBeTruthy();
});

test("Tooltip renders with custom id", () => {
  const { getByRole } = render(() => <Tooltip content="Test content" id="test-id">
    <button>Hover me</button>
  </Tooltip>);

  const tooltipElement = getByRole("tooltip");
  expect(tooltipElement).toHaveAttribute("aria-describedby", "test-id");
});

test("Tooltip renders with custom content", async () => {
  render(() => <Tooltip content={<span>Test content</span>}>
    <button>Hover me</button>
  </Tooltip>);

  const triggerElement = screen.getByRole("tooltip");
  triggerElement.focus();

  const contentElement = screen.getByText("Test content");
  expect(contentElement).toBeTruthy();
});

test("Tooltip renders with custom trigger", () => {
  const { getByRole } = render(() => <Tooltip content={<span>Test content</span>}>
    <span>Hover me</span>
  </Tooltip>);

  const triggerElement = getByRole("tooltip");
  expect(triggerElement).toHaveTextContent("Hover me");
});

test("Tooltip snapshot test", () => {
  const { asFragment } = render(() => <Tooltip content="Test content">
    <button>Hover me</button>
  </Tooltip>);
  expect(asFragment()).toMatchSnapshot();
});
