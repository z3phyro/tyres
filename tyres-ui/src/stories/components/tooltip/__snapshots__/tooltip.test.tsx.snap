// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip snapshot test 1`] = `"<button data-closed="" role="tooltip" class="focus:ring-2 ring-offset-2 ring-blue-500 outline-none rounded-sm"><button>Hover me</button></button>"`;
